.github/workflows/ci.yml

```
@@ -50,12 +50,14 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: Install dependencies
-      run: |
-        sudo apt install verilator libelf-dev
-        pip install -r python-requirements.txt
+    - name: Install Nix
+      uses: cachix/install-nix-action@v24
 
     - name: Build Verilator simulator
       run: |
-        fusesoc --cores-root=. run --target=sim --tool=verilator --setup --build lowrisc:sonata:system --verilator_options="+define+RVFI"
-
+        # TODO: Actually build with nix so this tests the sonata-simulator
+        # package build
+        nix develop . -c fusesoc --cores-root=. run --target=sim \
+          --tool=verilator --setup --build lowrisc:sonata:system \
+          --verilator_options="+define+RVFI -j $NIX_BUILD_CORES" \
+          --make_options="-j $NIX_BUILD_CORES"
```

```suggestion
        nix build .#sonata-simulator
```

This'll build the simulator in a sandbox

---

resolved, I believe.